src/linear_algebra/trace.lean

```diff
@@ -96,4 +98,25 @@ if H : ∃ (s : finset M), nonempty (basis s R M) then let ⟨s, ⟨b⟩⟩ := H
 by { simp_rw [trace_eq_matrix_trace R b, linear_map.to_matrix_mul], apply matrix.trace_mul_comm }
 else by rw [trace, dif_neg H, linear_map.zero_apply, linear_map.zero_apply]
 
+/-- The trace of an endomorphism is invariant under conjugation -/
+@[simp]
+theorem trace_conj (f g : M →ₗ[R] M) [invertible f] :
+  trace R M (f * g * ⅟ f ) = trace R M g :=
+by { rw trace_mul_comm, simp }
+
+end
+
+section
+variables (R : Type u) [field R] {M : Type v} [add_comm_group M] [module R M]
+open_locale classical
```

Comment:
What does Lean complain about if you leave this out?

Reply:
`finrank_eq_card_finset_basis` needs R to be a `division_ring`